src/gluonts/dataset/util.py

```diff
@@ -132,11 +126,15 @@ def to_pandas(instance: dict, freq: str = None) -> pd.Series:
     pandas.Series
         Pandas time series object.
     """
-    target = instance["target"]
-    start = instance["start"]
-    if not freq:
-        freq = start.freqstr
-    index = pd.date_range(start=start, periods=len(target), freq=freq)
+    target = instance[FieldName.TARGET]
+    start = instance[FieldName.START]
+
+    if FieldName.INDEX in instance:
+        index = instance[FieldName.INDEX]
```

Comment:
yes sure, both can be provided and are used actually since I get the freq sometimes from the start datetime rather than the datetime index which doesn't have a freq in the case of irregular dates.